### ISSUE TITLE
Entity PATCH deletes attributes unexpectedly

### ISSUE THREAD

**Issue by jason-fox** [Fiware/context.Orion-LD#506]
I have an Entity as shown:

```jsonld
{
    "@context": "http://context-provider:3000/data-models/ngsi-context.jsonld",
    "id": "urn:ngsi-ld:TemperatureSensor:001",
    "type": "TemperatureSensor",
    "category": {
        "type": "Property",
        "value": "sensor"
    },
    "temperature": {
        "type": "Property",
        "value": 25,
        "unitCode": "CEL"
    },
    "batteryLevel": {
        "value": 0.9,
        "type": "Property",
        "unitCode": "C62"
    },
    "controlledAsset": {
        "object": "urn:ngsi-ld:Building:barn002",
        "type": "Relationship"
    }
}
```

I wish to update multiple attributes (in this case `category` and `controlledAsset`:

```console
curl -L -X PATCH 'http://localhost:1026/ngsi-ld/v1/entities/urn:ngsi-ld:TemperatureSensor:001/attrs' \
-H 'Content-Type: application/json' \
-H 'Link: <http://context-provider:3000/data-models/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"' \
--data-raw '{
      "category": {
            "value": [
                  "sensor",
                  "actuator"
            ],
            "type": "Property"
      },
      "controlledAsset": {
            "type": "Relationship",
            "object": "urn:ngsi-ld:Building:barn001"
      }
}'
```

Expected response **200 Success** actual response **207 Multistatus**

```jsonld
{
    "@context": "http://context-provider:3000/data-models/ngsi-context.jsonld",
    "updated": [
        "https://uri.fiware.org/ns/data-models#category",
        "https://uri.fiware.org/ns/data-models#controlledAsset"
    ],
    "notUpdated": [
        {
            "attributeName": "https://w3id=org/saref#temperature",
            "reason": "Error"
        },
        {
            "attributeName": "https://uri=fiware=org/ns/data-models#batteryLevel",
            "reason": "Error"
        }
    ]
}
```

Furthermore  `temperature` and `batteryStatus` attributes are **deleted*

According to section **6.6.3.2 PATCH**, the payload sent is an **NGSI-LD Entity Fragment**

- NGSI-LD Entity Fragment - Entity Fragment containing a complete representation of the Attributes to be updated. 

since neither  `temperature` nor `batteryStatus` was supplied they should be unaltered

```
"git_hash": "db891a72a531ef4cc037caed5b04f8fe71da6814",
"compile_time": "Thu Jul 30 13:13:23 UTC 2020"
```

**Comment by kzangeli** [comment]
Fixed by PR #514

**Comment by jason-fox** [comment]
verified eddd37530fb7cb748ad582b5c20ea4327accd558
